### PR TITLE
made patient notes consultation specific

### LINKED ISSUE #1210: Consultation Specific Doctor Notes
**Is your feature request related to a problem? Please describe.**
Currently the `doctor_notes` feature is patient specific, and not consultation/encounter specific.

**Describe the solution you'd like**
Add an optional FK relation field `consultation_id` in the `patient_notes` table to link certain notes to a consultation

**TODO**

- [ ] Create the Frontend issues to use Consultation Specific Notes when viewing notes from inside a consultation

### PR DESCRIPTION
**Pull request by Bhavik-ag**
## Proposed Changes

- Added new field `consultation` in patient notes model
- Added filter in patient notes based on consultation external id
- Added tests for the same

### Associated Issue

fixed #1210 

@coronasafe/care-backend-maintainers @coronasafe/care-backend-admins

### CODE REVIEW

**Comment by Bhavik-ag** [comment]
@sainak please review this  PR

File: care/facility/api/viewsets/patient.py [hunk 1]
```diff
@@ -640,8 +646,14 @@ def perform_create(self, serializer):
             raise ValidationError(
                 {"patient": "Only active patients data can be updated"}
             )
+
+        consultation = None
+        if self.request.data.get("consultation"):
+            consultation = patient.last_consultation
+
```
Comment (sainak) [inline care/facility/api/viewsets/patient.py:653]: this is not the way to handle optional fields

Comment (sainak) [reply care/facility/api/viewsets/patient.py:653]: use serializers

File: care/facility/api/viewsets/patient.py [hunk 2]
```diff
@@ -640,8 +646,14 @@ def perform_create(self, serializer):
...
         return serializer.save(
             facility=patient.facility,
             patient=patient,
+            consultation=consultation,
```
Comment (sainak) [inline care/facility/api/viewsets/patient.py:657]: to be handled in serializers

**Review by sainak — CHANGES_REQUESTED** [review]

**Review by rithviknishad — APPROVED** [review]

File: care/facility/models/patient.py [hunk 3]
```diff
@@ -680,6 +680,9 @@ class PatientNotes(FacilityBaseModel, PatientRelatedPermissionMixin):
     patient = models.ForeignKey(
         PatientRegistration, on_delete=models.PROTECT, null=False, blank=False
     )
+    consultation = models.ForeignKey(
+        PatientConsultation, on_delete=models.PROTECT, null=True, blank=True
+    )
```
Comment (rithviknishad) [inline care/facility/models/patient.py:685]: @sainak should we index this field?

Comment (sainak) [reply care/facility/models/patient.py:685]: Index for FK is enabled by default

Comment (rithviknishad) [reply care/facility/models/patient.py:685]: owuuu oki

**Pull request closed without merging**